.chronus/changes/witemple-msft-visibility-1_0-2025-1-20-17-25-43.md

```diff
@@ -0,0 +1,7 @@
+---
+changeKind: deprecation
+packages:
+  - "@typespec/compiler"
+---
+
+Deprecate use of string-based visibility modifiers using warnings.
```

Comment:
can you add some diff code block to show how to migrate

Reply:
Yes. I think I'm also going to add a codefix if all the strings in the args are lifecycle strings.

Reply:
Resolving. I implemented a codefix (it automatically fixes `@visibility(string)` and `@visibility("none")`, everything else is manual) and added some examples to this entry with some words about how to convert.